src/Analyser/ExpressionTypeHolder.php

```diff
@@ -35,15 +35,15 @@ public function equals(self $other): bool
 
 	public function and(self $other): self
```

Comment:
this method is on the hot path, prevent unncessary function/method call overhead